packages/turf-intersect/index.js

```diff
@@ -34,22 +35,17 @@ var jsts = require('jsts');
  * var addToMap = [poly1, poly2, intersection];
  */
 module.exports = function (poly1, poly2) {
-    var geom1, geom2;
-    if (poly1.type === 'Feature') geom1 = poly1.geometry;
-    else geom1 = poly1;
-    if (poly2.type === 'Feature') geom2 = poly2.geometry;
-    else geom2 = poly2;
+    var geom1 = (poly1.type === 'Feature') ? poly1.geometry : poly1;
+    var geom2 = (poly2.type === 'Feature') ? poly2.geometry : poly2;
+
     var reader = new jsts.io.GeoJSONReader();
-    var a = reader.read(JSON.stringify(geom1));
-    var b = reader.read(JSON.stringify(geom2));
+    var a = reader.read(JSON.stringify(truncate(geom1)));
+    var b = reader.read(JSON.stringify(truncate(geom2)));
```

Comment:
Since `truncate` does the `clone` operation, we can drop `JSON.stringify` from the mix.